src/slurm_plugin/clustermgtd.py

```diff
@@ -440,47 +434,33 @@ def manage_cluster(self):
             ComputeFleetStatus.PROTECTED,
         }:
             # Get node states for nodes in inactive and active partitions
-            try:
-                log.info("Retrieving nodes info from the scheduler")
-                partitions = {
-                    partition.name: partition for partition in ClusterManager._get_partition_info_with_retry()
-                }
-                log.debug("Partitions: %s", partitions)
-                active_nodes, inactive_nodes = self._get_node_info_from_partition(partitions)
-                log.debug("Current active slurm nodes in scheduler: %s", active_nodes)
-                log.debug("Current inactive slurm nodes in scheduler: %s", inactive_nodes)
-                self._inactive_partitions = get_inactive_partition_names(partitions)
-            except ClusterManager.SchedulerUnavailable:
-                log.error("Unable to get partition/node info from slurm, no other action can be performed. Sleeping...")
+            active_nodes, inactive_nodes, self._partitions_name_map = self._initialize_nodes()
+            if None in [active_nodes, inactive_nodes, self._partitions_name_map]:
                 return
+
             # Get all non-terminating instances in EC2
-            try:
-                # After reading Slurm nodes wait for 5 seconds to let instances appear in EC2 describe_instances call
-                time.sleep(5)
-                log.info("Retrieving list of EC2 instances associated with the cluster")
-                cluster_instances = self._get_ec2_instances()
-            except ClusterManager.EC2InstancesInfoUnavailable:
-                log.error("Unable to get instances info from EC2, no other action can be performed. Sleeping...")
+            cluster_instances = self._get_cluster_instances()
+            if cluster_instances is None:
                 return
-            # Initialize mapping data structures
-            ip_to_slurm_node_map = {node.nodeaddr: node for node in active_nodes}
-            log.debug("ip_to_slurm_node_map %s", ip_to_slurm_node_map)
+
+            log.debug("Current cluster instances in EC2: %s", cluster_instances)
+
+            # Initialize nodes
+            instance_id_to_active_node_map, inactive_instance_ids, orphaned_instances = self._map_instances_and_nodes(
```

Comment:
Can we get rid of all these extra data structures and hold this information in either Partition or Node objects?

What I have in mind is the following: after loading the ec2 instances we could store this info as part of a Slurm Node object. This way we can simply pass the list of Slurm partitions around:
* To get the list of inactive/active nodes you just get nodes from all inactive/active partitions. 
* The partition object will have a method to get the list of orphaned_instances.
* you will not need inactive_instance_ids anymore since you can retrieve this from partition.node.instance.id

At a high level the `manage_cluster` function could look like:
```
...
nodes = self._retrieve_scheduler_nodes()
partitions = self._retrieve_scheduler_partitions(nodes)
ec2_instances = self._get_ec2_instances()  # ec2_instances rather than a plain list could be a map(private_ip, instance)
self._update_slurm_nodes_with_ec2_info(nodes, ec2_instances)  # here we could set the node.ec2_instance property for each Slurm node

# Handle inactive partition and terminate backing instances
self._clean_up_inactive_partition(partitions)
if not self._config.disable_all_health_checks:
    self._perform_health_check_actions(partitions)
# Maintain slurm nodes
self._maintain_nodes(partitions)
# Clean up orphaned instances
self._terminate_orphaned_instances(partitions)  
```